hera_mc/correlator.py

```diff
@@ -268,6 +268,284 @@ def _get_config(corr_cm=None, correlator_redis_address=DEFAULT_REDIS_ADDRESS):
+        )
+
+
+def _parse_config(config, config_hash):
+    """
+    Parse the config into all the components required for the various config tables.
+
+    Parameters
+    ----------
+    config : dict
+        Dict of config info from redis/hera_corr_cm.
+    config_hash : str
+        Unique hash of the config.
+
+
+    Returns
+    -------
+    list
+        List of correlator config related objects including CorrelatorConfigParams,
+        CorrelatorConfigActiveSNAP, CorrelatorConfigInputIndex, and
+        CorrelatorConfigPhaseSwitchIndex objects.
+
+    """
+    obj_list = []
+    keys_to_save = ['fft_shift', 'fpgfile', 'dest_port', 'log_walsh_step_size',
+                    'walsh_order', 'walsh_delay']
+    for key in keys_to_save:
+        value = config[key]
+        obj_list.append(
+            CorrelatorConfigParams.create(
+                config_hash=config_hash, parameter=key, value=value
+            )
+        )
+
+    fengines = sorted(config['fengines'].keys())
+    obj_list.append(
+        CorrelatorConfigParams.create(
+            config_hash=config_hash, parameter='fengines', value=','.join(fengines)
+        )
+    )
+    xengines = [str(x) for x in sorted(config['xengines'].keys())]
+    obj_list.append(
+        CorrelatorConfigParams.create(
+            config_hash=config_hash,
+            parameter='xengines',
+            value=','.join(xengines)
+        )
+    )
+    for xengind, xeng in config['xengines'].items():
+        parameter = f'x{xengind}:chan_range'
+        value = ','.join([str(_x) for _x in xeng['chan_range']])
+        obj_list.append(
+            CorrelatorConfigParams.create(
+                config_hash=config_hash,
+                parameter=parameter,
+                value=value
+            )
+        )
+        for evod in ['even', 'odd']:
+            for ipma in ['ip', 'mac']:
+                parameter = f'x{xengind}:{evod}:{ipma}'
+                value = xeng[evod][ipma]
+                obj_list.append(
+                    CorrelatorConfigParams.create(
+                        config_hash=config_hash,
+                        parameter=parameter,
+                        value=value
+                    )
+                )
+    for hostname, entry in config['fengines'].items():
+        obj_list.append(
+            CorrelatorConfigActiveSNAP.create(
+                config_hash=config_hash,
+                hostname=hostname,
+            )
+        )
+        for antind in range(len(entry['ants'])):
+            cind = entry['ants'][antind]
+            obj_list.append(
+                CorrelatorConfigInputIndex.create(
+                    config_hash=config_hash,
+                    correlator_index=cind,
+                    hostname=hostname,
+                    antenna_index_position=antind,
+                )
+            )
+        for psind, pind in enumerate(entry['phase_switch_index']):
+            pind = entry['phase_switch_index'][psind]
```

Comment:
If you're going to immediately overwrite the value of `pind`, it's better to make it just an underscore (`_`) in the `enumerate` construction. So something like:
```
    for psind, _ in enumerate(entry["phase_switch_index"]):
        pind = entry["phase_switch_index"][psind]
```
At that point, though, it seems like you really only need an index, so you could do something like:
```
    for psind in range(len(entry["phase_switch_index"]):
        pind = entry["phase_switch_index"][psind]
```